spec/lib/pdf_filler/nj1040_pdf_spec.rb

```diff
@@ -197,5 +208,53 @@
         expect(pdf_fields["ZIP Code"]).to eq "08037"
       end
     end
+
+    describe "filing status" do
+      context "single" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 1
+        end
+        it "checks the appropriate box" do
+          expect(pdf_fields["Group1"]).to eq "Choice1"
+        end
+      end
+
+      context "married filing jointly" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 2
+        end
+        it "checks the appropriate box" do
+          expect(pdf_fields["Group1"]).to eq "Choice2"
+        end
+      end
+
+      context "married filing separately" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 3
+        end
+        it "checks the appropriate box" do
+          expect(pdf_fields["Group1"]).to eq "Choice3"
+        end
+      end
+
+      context "head of household" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 4
+        end
+        it "checks the appropriate box" do
+          expect(pdf_fields["Group1"]).to eq "Choice4"
+        end
+      end
+
+      context "qualifying widow" do
```

Comment:
[dust/pebble] I don't see year of spouse death in the PDF part of this code, but I see it in the XML. Should that be in scope for this story if it's in the XML scope?

Reply:
Wow, yes, ty for catching!